deeppavlov/models/sklearn/sklearn_component.py

```diff
@@ -0,0 +1,357 @@
+        except:
+            try:
+                if issparse(x_features):
+                    log.info("Converting input for model {} to dense array".format(self.model_name))
+                    predictions = getattr(self.model, self.infer_method)(x_features.todense())
+                else:
+                    log.info("Converting input for model {} to sparse array".format(self.model_name))
+                    predictions = getattr(self.model, self.infer_method)(csr_matrix(x_features))
+            except:
+                raise ConfigError("Can not infer on the given data".format(self.model_name))
+
+        if isinstance(predictions, list):
+            #  ``predict_proba`` sometimes returns list of n_outputs (each output corresponds to a label)
+            #  but we will return (n_samples, n_labels)
+            #  where each value is a probability of a sample to belong with the label
+            predictions_ = [[predictions[j][i][1] for j in range(len(predictions))] for i in range(x_features.shape[0])]
+            predictions = np.array(predictions_)
+
+        if len(predictions.shape) == 1:
+            predictions = predictions.reshape(-1, 1)
+        return predictions
+
+    def init_from_scratch(self) -> None:
+        """
+        Initialize ``self.model`` as some sklearn model from scratch with given in ``self.model_params`` parameters.
+
+        Returns:
+            None
+        """
+        log.info("Initializing model {} from scratch".format(self.model_name))
+        model_function = cls_from_str(self.model_name)
+
+        if model_function is None:
+            raise ConfigError("Model with {} model_name was not found.".format(self.model_name))
+
+        given_params = {}
+        if self.model_params:
+            available_params = self.get_function_params(model_function)
+            for param_name in self.model_params.keys():
+                if param_name in available_params:
+                    try:
+                        given_params[param_name] = cls_from_str(self.model_params[param_name])
+                    except:
+                        given_params[param_name] = self.model_params[param_name]
+
+        self.model = model_function(**given_params)
+        return
+
+    def load(self, fname: str = None) -> None:
+        """
+        Initialize ``self.model`` as some sklearn model from saved re-initializing ``self.model_params`` parameters. \
+            If in new given parameters ``warm_start`` is set to True and given model admits ``warm_start`` parameter, \
+            model will be initilized from saved with opportunity to continue fitting.
+
+        Args:
+            fname: string name of path to model to load from
+
+        Returns:
+            None
+        """
+        if fname is None:
+            fname = self.load_path
+
+        if Path(fname).suffix != ".pkl":
+            fname = str(Path(fname).stem) + ".pkl"
+
+        if fname.exists():
```

Comment:
if `fname` is `str`, this line will raise

Reply:
fixed